catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/MgrsCoordinateProcessor.java

```diff
@@ -0,0 +1,355 @@
+import org.slf4j.LoggerFactory;
+
+/**
+ * A regex-based processor for identifying potential MGRS coordinate literals in a given input
+ * string despite deviations from the official spec. The general policy is to be as forgiving as
+ * possible without sacrificing determinism.
+ *
+ * <p>Due to the strict expectations of usng4j regarding input strings, some additional processing
+ * is done to "backfill" necessary zeros for occurrences of a {@link MgrsPartType#NUMERIC} string.
+ *
+ * @see #processNumericType(String) for more information on this "backfilling" behavior.
+ */
+@SuppressWarnings("squid:S1135" /* Ticket number provided on comment */)
+public class MgrsCoordinateProcessor {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MgrsCoordinateProcessor.class);
+
+  private static final Pattern PATTERN_MGRS_ZONE = Pattern.compile("\\d\\d?[CDEFGHJKLMNPQRSTUVWX]");
+
+  private static final Pattern PATTERN_MGRS_100KM =
+      Pattern.compile("[ABCDEFGHJKLMNPQRSTUVWXYZ][ABCDEFGHJKLMNPQRSTUV]");
+
+  private static final Pattern PATTERN_MGRS_NUMERIC =
+      Pattern.compile("((\\d){1,5}(\\h)+(\\d){1,5}(\\h)+)|((\\d){2,10})");
+
+  private static final Pattern PATTERN_MGRS_ANYPART =
+      Pattern.compile(
+          format(
+              "(%s)|(%s)|(%s)",
+              PATTERN_MGRS_ZONE.pattern(),
+              PATTERN_MGRS_100KM.pattern(),
+              PATTERN_MGRS_NUMERIC.pattern()));
+
+  // usng4j helper constant for backfilling zeros
+  private static final Integer MAX_EASTING_LENGTH = 5;
+
+  // usng4j helper constant for backfilling zeros
+  private static final Integer MAX_NORTHING_LENGTH = 5;
+
+  // This key tells the UI that the geo is on the suggestion itself
+  private static final String LITERAL = "LITERAL2";
+
+  /**
+   * An {@link MgrsPartType} identifies one of the three different component strings that are glued
+   * together to form a valid MGRS coordinate string, including a category {@link #NONE} to be used
+   * when the string is unsuitable for any part of an MGRS coordinate or the type has not been
+   * provided yet.
+   *
+   * <p>Each type corresponds to a regex defined above.
+   */
+  private enum MgrsPartType {
+    NONE,
+    ZONE,
+    HUNDRED_KM,
+    NUMERIC
+  }
+
+  private final CoordinateSystemTranslator translator = new CoordinateSystemTranslatorImpl();
+
+  /**
+   * Given a list of {@link Suggestion}s and the query that yielded them, enhance the list with
+   * additional suggestions based upon the presence of coordinate literals in the query string.
+   *
+   * @param results the list of results to enhance.
+   * @param query the query string to search for coordinate literals.
+   * @return the enhanced list.
+   */
+  public List<Suggestion> enhanceResults(List<Suggestion> results, String query) {
+    LiteralSuggestion literal = getMgrsSuggestion(query);
+    if (literal != null && literal.hasGeo()) {
+      LOGGER.debug("Adding the MGRS suggestion to results");
```

Comment:
✏️ `trace` is a more appropriate debug level

Reply:
There is an overarching decision to be made by this object: _Do I add an MGRS suggestion to the list?_

By moving this to `TRACE`, If `DEBUG` is selected for this package, there would be no clear indication of this top level decision.

Reply:
Trace
-This level is for tracing program operation
-Avoid stack traces at this level

Debug
- Most standard exception logging should be done at this level
- Stack traces should be at this level

Reply:
Unless this is an error state trace is more appropriate

Reply:
this seems more appropriate as trace

Reply:
I'm happy to change it but I think we are confusing the following: 
Stack traces should be at this level
ONLY stack traces should be at this level

The document doesn't forbid stuff like this. 
https://codice.atlassian.net/wiki/spaces/DDF/pages/7045131/Log+Levels

I'll change it but we probably want more clarification on the document.

Reply:
Fixed.